django/db/models/fields/related.py

```diff
@@ -615,6 +615,16 @@ def _check_unique_target(self):
         if not self.foreign_related_fields:
             return []
 
+        # If a model defines Meta.primary_key and a foreign key refers to it,
+        # the check should be skipped (since primary keys are unique).
+        pk = self.remote_field.model._meta.primary_key
+        if pk:
+            pk = set(pk)
+            if pk == {f.attname for f in self.foreign_related_fields}:
+                return []
+            elif pk == {f.name for f in self.foreign_related_fields}:
+                return []
```

Comment:
So this was because I wasn't sure if we want to allow setting primary keys on foreign key fields, for example. I think we shouldn't, I added a system check for this. So it should be enough to check for the attnames.

`set(pk)` here is the same as `Meta.primary_key`, it's tuple or list of strings.

Reply:
I changed this using a different approach @charettes suggested. I'm resolving this thread, please open a new one if there's an issue with the new approach.